visualization/app/codeCharta/ui/legendPanel/legendScreenshotButton/legendScreenshotButton.component.ts

```diff
@@ -0,0 +1,93 @@
+import { screenshotToClipboardEnabledSelector } from "../../../../../app/codeCharta/state/store/appSettings/enableClipboard/screenshotToClipboardEnabled.selector"
+import { getVisibleFileStates, isDeltaState, isPartialState } from "../../../../../app/codeCharta/model/files/files.helper"
+
+@Component({
+	selector: "cc-legend-screenshot-button",
+	templateUrl: "./legendScreenshotButton.component.html",
+	styleUrls: ["./legendScreenshotButton.component.scss"],
+	encapsulation: ViewEncapsulation.None
+})
+export class LegendScreenshotButtonComponent {
+	@Input() isLegendVisible: boolean
+	isRenderingScreenshot: boolean
+	isWriteToClipboardAllowed: boolean = "write" in navigator.clipboard
+	isScreenshotToClipboardEnabled$ = this.store.select(screenshotToClipboardEnabledSelector)
+
+	constructor(
+		private store: Store<CcState>,
+		private state: State<CcState>,
+		public isAttributeSideBarVisibleService: IsAttributeSideBarVisibleService
+	) {}
+
+	async makeScreenshotToFile() {
+		this.isRenderingScreenshot = true
+		const element = this.getLegendPanelHTMLElement()
+		const canvas = await this.captureScreenshot(element)
+		const dataUrl = canvas.toDataURL("image/png")
+		this.downloadScreenshot(dataUrl, this.makePNGFileName())
+		this.isRenderingScreenshot = false
+	}
+
+	async makeScreenshotToClipboard() {
+		this.isRenderingScreenshot = true
+		const element = this.getLegendPanelHTMLElement()
+		const canvas = await this.captureScreenshot(element)
+		const blob: Blob = await new Promise(resolve => canvas.toBlob(resolve))
+		await this.setToClipboard(blob)
+		this.isRenderingScreenshot = false
+	}
+
+	getLegendPanelHTMLElement() {
+		return document.getElementById("legend-panel")
+	}
+
+	async captureScreenshot(element: HTMLElement): Promise<HTMLCanvasElement> {
+		const tagsNamesToIgnore = new Set([
+			"cc-tool-bar",
+			"cc-file-extension-bar",
+			"cc-ribbon-bar",
+			"cc-code-map",
+			"cc-attribute-side-bar",
+			"cc-loading-file-progess-spinner"
+		])
+		return html2canvas(element, {
+			removeContainer: true,
+			scale: 2,
+			allowTaint: true,
+			ignoreElements(element) {
+				return tagsNamesToIgnore.has(element.tagName.toLowerCase())
+			}
+		})
+	}
+
+	downloadScreenshot(dataUrl: string, fileName: string): void {
+		const downloadLink = document.createElement("a")
+		downloadLink.download = fileName
+		downloadLink.href = dataUrl
+		document.body.appendChild(downloadLink)
+		downloadLink.click()
+		downloadLink.remove()
+	}
+
+	async setToClipboard(blob: Blob) {
+		const data = [new ClipboardItem({ [blob.type]: blob })]
+		await navigator.clipboard.write(data)
+	}
+
+	makePNGFileName(): string {
+		const files = this.state.getValue().files
+		const jsonFileNames = getVisibleFileStates(files)
+		const state = isPartialState(files) ? "partial" : isDeltaState(files) ? "delta" : ""
+		const pngFileName = `${jsonFileNames
+			.map(fileState => fileState.file.fileMeta.fileName.slice(0, fileState.file.fileMeta.fileName.indexOf(".json")))
```

Comment:
The filename could get too long if there are many maps active. I am not certain how to do it properly, so that we have a good name but we should think about a filename length limit.

Reply:
What about e.g., `file.jsonstring_inside.json`? That would not be handled correct.

```suggestion
			.map(fileState => fileState.file.fileMeta.fileName.slice(0, fileState.file.fileMeta.fileName.lastIndexOf(".json")))
```

The extension could theoretically also be a different one than json later. But I guess it's a case that is unlikely.

Reply:
> The filename could get too long if there are many maps active. I am not certain how to do it properly, so that we have a good name but we should think about a filename length limit.


The method used to generate the file name for the legend screenshot was adapted from the one used for the map screenshot (although both are unified now). As this implementation has been problem-free for a some time, it seems unlikely to me to pose any issues.

Reply:
I think there haven't been any issues yet with too long filenames because the feature is not frequently used.
It is a common use case to uploaded lots of maps in the visualization. Most likely there will be issues soon when the feature is used more often.

Reply:
Resolved this conversion by accident :| I'll think about how to truncate the filename in a reasonable way.